sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/SessionTokenMismatchRetryPolicy.java

```diff
@@ -102,11 +107,27 @@ public Mono<ShouldRetryResult> shouldRetry(Exception e) {
                     this.maximumBackoff);
         }
 
+        // For remote region preference ensure that the last retry is long enough (even when exceeding max backoff time)
+        // to consume the entire minRetryTimeInLocalRegion
+        if (sessionRetryOptionsAccessor.getRegionSwitchHint(sessionRetryOptions) ==
+                CosmosRegionSwitchHint.REMOTE_REGION_PREFERRED
+            && attempt >= (this.maxRetryAttemptsInCurrentRegion.get() - 1)) {
+
+            Duration remainingMinRetryTimeInLocalRegion = this.waitTimeTimeoutHelper.getRemainingTime(
+                sessionRetryOptionsAccessor.getMinInRegionRetryTime(this.sessionRetryOptions)
```

Comment:
possible negative value?

Reply:
Not a problem - will only ever be used when it is larger than the effectiveBackoff